Src/FluentAssertions/Specialized/NonGenericAsyncFunctionAssertions.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Specialized
+{
+    public class NonGenericAsyncFunctionAssertions : AsyncFunctionAssertions
+    {
+        private readonly ITimer timer;
+
+        public NonGenericAsyncFunctionAssertions(Func<Task> subject, IExtractExceptions extractor, ITimer timer) : base(subject,
+            extractor)
+        {
+            this.timer = timer;
+        }
+
+        /// <summary>
+        /// Asserts that the current <see cref="Task{T}"/> will complete within specified time.
+        /// </summary>
+        /// <param name="timeSpan">The allowed time span for the operation.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndWhichConstraint<AsyncFunctionAssertions, Task> CompleteWithin(
```

Comment:
If the other `CompleteWithin` is changed to 
`AndWhichConstraint<GenericAsyncFunctionAssertions<TResult>, TResult>` 
then this should be changed to
AndConstraint<AsyncFunctionAssertions<TResult>>

Reply:
I don't get it. They both return `AndWhichConstraint`.  One just wraps it in `Task`.